Realm.Shared/native/SortDescriptorBuilder.cs

```diff
@@ -0,0 +1,87 @@
+﻿////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2016 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+
+namespace Realms.native
+{
+    internal class SortDescriptorBuilder
+    {
+        public class Clause
+        {
+            public List<IntPtr> ColumnIndexChain;
+            public bool Ascending;
+
+            [StructLayout(LayoutKind.Sequential)]
+            public struct Marshalable
+            {
+                public IntPtr Offset;
+                public IntPtr Count;
+
+                [MarshalAs(UnmanagedType.I1)]
+                public bool Ascending;
+            }
+        }
+
+        private readonly List<Clause> _clauses = new List<Clause>();
+        private readonly RealmObject.Metadata _metadata;
+
+        public TableHandle TableHandle => _metadata.Table;
+
+        public SortDescriptorBuilder(RealmObject.Metadata metadata)
+        {
+            _metadata = metadata;
+        }
+
+        // This method adds a direct sort clause. To add clauses that follow links, 
```

Comment:
Yeah, that *is* unclear, true. It's just to say that this method currently doesn't support creating an array of indices but just one level. Once we want to support sorting via links, we will have to add a method that takes a collection of indices instead of just one. I'll change it.